lib/usd/translators/meshWriter_BlendShapes.cpp

```diff
@@ -0,0 +1,960 @@
+//
+// Copyright 2020 Apple
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "jointWriter.h"
+#include "meshWriter.h"
+
+#include <mayaUsd/fileio/primWriter.h>
+#include <mayaUsd/fileio/translators/translatorUtil.h>
+#include <mayaUsd/fileio/utils/meshWriteUtils.h>
+#include <mayaUsd/fileio/utils/writeUtil.h>
+
+#include <pxr/base/tf/diagnostic.h>
+#include <pxr/base/tf/stringUtils.h>
+#include <pxr/base/vt/types.h>
+#include <pxr/usd/sdf/path.h>
+#include <pxr/usd/usdGeom/pointBased.h>
+#include <pxr/usd/usdSkel/bindingAPI.h>
+#include <pxr/usd/usdSkel/blendShape.h>
+
+#include <maya/MAnimUtil.h>
+#include <maya/MApiNamespace.h>
+#include <maya/MFloatArray.h>
+#include <maya/MFloatPointArray.h>
+#include <maya/MFnAttribute.h>
+#include <maya/MFnBlendShapeDeformer.h>
+#include <maya/MFnComponentListData.h>
+#include <maya/MFnGeometryFilter.h>
+#include <maya/MFnPointArrayData.h>
+#include <maya/MFnSingleIndexedComponent.h>
+#include <maya/MGlobal.h>
+#include <maya/MItDependencyGraph.h>
+#include <maya/MObject.h>
+#include <maya/MPointArray.h>
+#include <maya/MStatus.h>
+
+#include <algorithm>
+#include <complex>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#define MAYA_BLENDSHAPE_EVAL_HOTFIX 1
+
+#define MAYA_ATTR_NAME_WEIGHT "weight"
+
+struct MayaBlendShapeWeightDatum
+{
+    MObjectArray targetMeshes; // The target shape(s) to hit. (i.e. multiple shapes would be using
+                               // Maya's "in-betweens" feature.)
+    MIntArray
+              inputTargetGroupIndices; // The input group indices at which each target is connected under.
```

Comment:
Looks like `clang-format` mangled this? It seems to handle comments better when they're on the line before rather than trailing like this.

Reply:
Yea, clang-format "adjusted" this. I can re-arrange the comment to see if that helps.